integration-notification-service/src/main/java/md/maib/integration/notification/service/controller/NotificationController.java

```diff
@@ -1,12 +1,29 @@
 package md.maib.integration.notification.service.controller;
 
+import md.maib.integration.notification.service.model.EventModel;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
 @Controller
 @RestController
 @RequestMapping("/api/notifications")
 public class NotificationController {
+    private final RabbitTemplate rabbitTemplate;
+
+    @Autowired
+    public NotificationController(RabbitTemplate rabbitTemplate) {
+        this.rabbitTemplate = rabbitTemplate;
```

Comment:
you can use use requareArgsConstructor

Reply:
![image](https://github.com/user-attachments/assets/4f15dd97-c61c-4721-ba9d-6a029d3556e5)
If I use @RequiredArgsConstructor, I get this type of error. I could use @AllArgsConstructor instead, but that would remove the final state of the fields. So, I decided to stick with the old option since I’m more used to that. 😄